autoload/ale/c.vim

```diff
@@ -192,7 +196,7 @@ function! ale#c#FindProjectRoot(buffer) abort
 
     " Fall back on detecting the project root based on other filenames.
     if empty(l:root)
-        for l:project_filename in g:__ale_c_project_filenames
+        for l:project_filename in g:ale_c_project_filenames
```

Comment:
The documentation says you can use `b:ale_c_project_filenames`, so make that true.

```suggestion
        for l:project_filename in ale#Var(a:buffer, 'c_project_filenames')
```